protovalidate/internal/extra_func.py

```diff
@@ -161,28 +161,15 @@ def is_email(string: celtypes.Value) -> celpy.Result:
 
 
 def is_uri(string: celtypes.Value) -> celpy.Result:
-    url = urlparse.urlparse(str(string))
-    # urlparse correctly reads the scheme from URNs but parses everything
-    # after (except the query string) as the path.
-    if url.scheme == "urn":
-        if not (url.path):
-            return celtypes.BoolType(False)
-    elif not all([url.scheme, url.netloc, url.path]):
-        return celtypes.BoolType(False)
-
-    # If the query string contains percent-encoding, then try to decode it.
-    # unquote will return the same string if it is improperly encoded.
-    if "%" in url.query:
-        return celtypes.BoolType(urlparse.unquote(url.query) != url.query)
-
-    return celtypes.BoolType(True)
+    """is_uri validates whether string is a valid URI."""
```

Comment:
That's not a very helpful comment. Is there a reason not to use the same doc as the source implementation?

```suggestion
    """Returns true if the string is an email address, for example "foo@example.com".

    URI is defined in the internet standard RFC 3986.
    Zone Identifiers in IPv6 address literals are supported (RFC 6874).
    """
```

- Gives an example so it's immediately clear what this is about.
- References the specification.
- Explains noteworthy details (zone id).

It's helpful information for users and maintainers.

Same for all other functions.

Reply:
I was actually copying docs from the Go implementation (which are just as unhelpful), so we should prob update that too.

Fixed.

Reply:
Yes, should update. 

The current version of `is_ip` raises an error if a version other than `4` or `6` is given. The reference implementation is documented with:

> Version 0 means either 4 or 6. Passing a version other than 0, 4, or 6 always returns false.

We want to nail down the behavior in those edge cases. That's why the docs are important.

Reply:
Ah yep. Good catch. I will update that in the forthcoming PR for porting `is_ip` logic.